src/aind_codeocean_utils/codeocean_job.py

```diff
@@ -0,0 +1,582 @@
+"""Module with generic Code Ocean job"""
+import logging
+import time
+from datetime import datetime
+from typing import Dict, List, Optional, Tuple, Union
+
+import requests
+from aind_codeocean_api.codeocean import CodeOceanClient
+from aind_codeocean_api.models.computations_requests import RunCapsuleRequest
+from aind_codeocean_api.models.data_assets_requests import (
+    CreateDataAssetRequest,
+    Source,
+    Sources,
+)
+
+LOG_FMT = "%(asctime)s %(message)s"
+LOG_DATE_FMT = "%Y-%m-%d %H:%M"
+
+logging.basicConfig(format=LOG_FMT, datefmt=LOG_DATE_FMT)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class CodeOceanJob:
+    """
+    This class contains convenient methods to register data assets,
+    run capsules, and capture results.
+    """
+
+    default_datetime_format = "%Y-%m-%d_%H-%M-%S"
+
+    def __init__(self, co_client: CodeOceanClient):
+        """
+        CapsuleJob class constructor.
+
+        Parameters
+        ----------
+        co_client : CodeOceanClient
+            A client that can be used to interface with the Code Ocean API.
+        """
+        self.co_client = co_client
+
+    def add_capture_time(self, name: str) -> str:
```

Comment:
I feel like this might be too specific. It's not really clear what the intention of this function is. It might be better to have the user pass in a custom lambda function or Callable that maps the input  data_asset name to an output name. I would try to avoid adding a datetime.now() call since it makes the function have different outputs given the same inputs.

Reply:
it depends on how high-level we want the function and what our use cases are in my opinion.

Reply:
I think though this function or something very similar is also in the aind_data_schema DataDescription class somewhere. I'll look into whether we can just import it.